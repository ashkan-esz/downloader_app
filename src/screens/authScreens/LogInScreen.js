import React from 'react';
import {View, StyleSheet, TouchableWithoutFeedback, Keyboard} from 'react-native';
import {Button, Text} from "react-native-elements";
import {Colors} from "../../styles";
import {LogInForm} from "../../components/organisms";
import {GoogleButton} from "../../components/atoms";
import {ScreenLayout} from "../../components/layouts";
import {useSelector} from "react-redux";

//todo : forget password
//todo : google sign in

const LogInScreen = ({navigation}) => {

    const internet = useSelector(state => state.user.internet);

    const paddingBottom = {
        paddingBottom: internet ? 0 : 38,
    }

    const googleSignIn = () => {
        //todo : add google login
    };

    return (
        <ScreenLayout>
            <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
                <View style={[style.container, paddingBottom]}>

                    {/*//todo : add logo*/}
                    <Text h3 style={style.header}>
                        <Text h3 style={{color: Colors.RED}}>
                            Welcome!
                        </Text>
                        <Text> </Text>
                        Login with your account and enjoy
                    </Text>

                    <LogInForm
                        extraStyle={style.loginForm}
                    />

                    <GoogleButton
                        extraStyle={style.googleButton}
                        text={"Or Login With"}
                        onPress={googleSignIn}
                    />

                    <View style={style.divider}/>
                    <Button
                        containerStyle={style.signInContainer}
                        titleStyle={style.signInText}
                        title={'CREATE ACCOUNT'}
                        type={"clear"}
                        onPress={() => {
                            navigation.navigate('SignUp');
                        }}
                    />

                </View>
            </TouchableWithoutFeedback>
        </ScreenLayout>
    );
};

const style = StyleSheet.create({
    container: {
        position: 'absolute',
        bottom: '2%',
        width: '85%',
        maxWidth: 350,
    },
    header: {
        color: '#b1aeae'
    },
    loginForm: {
        marginTop: 10
    },
    googleButton: {
        marginTop: 25
    },
    divider: {
        marginTop: 15,
        borderBottomColor: '#636161',
        borderBottomWidth: 0.9,
    },
    signInContainer: {
        marginTop: 12,
        alignSelf: 'center'
    },
    signInText: {
        color: Colors.RED
    }
})


export default LogInScreen;
