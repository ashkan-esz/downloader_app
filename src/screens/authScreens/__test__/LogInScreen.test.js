import * as React from "react";
import { fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-dom';
import LogInScreen from "../LogInScreen";

jest.mock("../../../components/organisms/LogInForm", () => {
        const {Button} = require("react-native");
        return ({onSubmit}) => {
            return (
                <Button title={'Login'} onPress={()=>onSubmit({title:'mocked login form'})}/>
            )
        }
    }
);


describe("Login/SignIn screen ", () => {

    test('renders correctly', () => {
        let {toJSON} = render(
            <LogInScreen/>
        );
        expect(toJSON()).toMatchSnapshot();
    });

    test('navigation to Home with form onsubmit', async () => {
        let navigate = jest.fn();
        let {getByText} = render(
            <LogInScreen navigation={{navigate}}/>
        )
        let signInButton = getByText('Login');
        expect(navigate).toHaveBeenCalledTimes(0);
        fireEvent.press(signInButton);
        expect(navigate).toHaveBeenCalledTimes(1);
        expect(navigate).toHaveBeenCalledWith('Home');
    });

    test('navigation to SignUp from "CREATE ACCOUNT" button', () => {
        let navigate = jest.fn();
        let {getByText} = render(
            <LogInScreen navigation={{navigate}}/>
        )
        let createAccButton = getByText('CREATE ACCOUNT');
        expect(navigate).toHaveBeenCalledTimes(0);
        fireEvent.press(createAccButton);
        expect(navigate).toHaveBeenCalledTimes(1);
        expect(navigate).toHaveBeenCalledWith('SignUp');
    });

});
