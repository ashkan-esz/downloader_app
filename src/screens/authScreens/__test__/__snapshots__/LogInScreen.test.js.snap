// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login/SignIn screen  renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1D1D27",
        "height": "100%",
        "width": "100%",
      },
      Object {
        "minHeight": 1334,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": "2%",
        "maxWidth": 350,
        "position": "absolute",
        "width": "85%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#b1aeae",
          "fontSize": 36,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#E15050",
            "fontSize": 36,
          }
        }
      >
        Welcome!
      </Text>
      <Text
        style={
          Object {
            "color": "#242424",
          }
        }
      >
         
      </Text>
      Login with your account and enjoy
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          google-btn
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#636161",
          "borderBottomWidth": 0.9,
          "marginTop": 15,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 3,
          "marginTop": 12,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#E15050",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            CREATE ACCOUNT
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
